### ISSUE TITLE
always crash

### ISSUE THREAD

**Issue by bianzhifeng** [rs/SDWebImage#2077]
### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [ ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / tvos
 Platform Version        | e.g. 8.0
 SDWebImage Version      | e.g. 3.7.6
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 7.3
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

I'm showing images on UICollectionView. I'm getting memory warning and app crashes after several run. 
i use leaks , find  decodeImageWithImage -> 238M .
help , thanks!

**Comment by bpoplauschi** [comment]
@bianzhifeng looks like this is the same issue with #586 and #538. Please take a look at those. Bottom line: decoding images is memory consuming, so if you are dealing with large images, you can use `SDWebImageScaleDownLargeImage` option. If your images are not that big, one option (https://github.com/rs/SDWebImage/issues/586#issuecomment-215059553) is to stop decompressing in the background
```
[[SDImageCache sharedImageCache] setShouldDecompressImages:NO];
[[SDWebImageDownloader sharedDownloader] setShouldDecompressImages:NO];
```
Please try using our latest version.

PS: Please fill in the info in the table above - so we know which version are you using and so on.

**Comment by bianzhifeng** [comment]
now, i update latest version 4.1.2.
i can't find setShouldDecompressImages in SDImageCache.

**Comment by bianzhifeng** [comment]
i will try this:        
        SDWebImageDownloader.shared().shouldDecompressImages = false
        SDImageCache.shared().config.shouldDecompressImages = false
Wish me luck.

**Comment by bpoplauschi** [comment]
@bianzhifeng sorry, for this version the configs from `SDImageCache` were moved to a separate object.
So `[SDImageCache sharedImageCache].config. shouldDecompressImages = NO;`

**Comment by bpoplauschi** [comment]
Yeap, your code is correct:
```
SDWebImageDownloader.shared().shouldDecompressImages = false
SDImageCache.shared().config.shouldDecompressImages = false
```

**Comment by bianzhifeng** [comment]
yes  i find , thank you very mach.
now  i will use : 
  SDWebImageDownloader.shared().downloadImage(with: URL(string: ""), options:        SDWebImageDownloaderOptions.scaleDownLargeImages, progress: nil) { [weak self] (downImg,
   data, error, finished) in
       self?.CollectionView?.reloadItems(at: [indexPath])
    } 
and give up
 sd_setImage(with: URL(string: ""), placeholderImage: UIImage(named: ""))
